src/data_hub/_api.py

```diff
@@ -183,12 +184,17 @@ def dynamic_function(
             )
 
         if not isinstance(response, dict):
-            response = response.json()
+            try:
+                response = response.json()
+            except JSONDecodeError:
+                response = dict(data=response.text())
```

Comment:
Fine for now, but a better solution would be to dispatch on the response `content-type` header.

Reply:
i should check for `'text/event-stream'`?

Reply:
looks like it's either `application/json` or `text/event-stream`?

Reply:
fixed